srml/executive/src/lib.rs

```diff
@@ -277,7 +279,8 @@ where
 				if index < &expected_index { internal::ApplyError::Stale } else { internal::ApplyError::Future }
 			) }
 			// pay any fees
-			Payment::make_payment(sender, encoded_len).map_err(|_| internal::ApplyError::CantPay)?;
+			let fee_multiplier = <system::Module<System>>::next_fee_multiplier();
```

Comment:
fees are not a concept that should be dealt with here. if it's a *weight multiplier* then rename as such. if it's really dealing with fees (i.e. balances) then it needs moving into a module that is aware of fees and balances.

Reply:
so at the moment (and thankfully given the time constraints) we are settling for the fact that _since weights are not generic over type_, __neither are fees__ (u32 for now; I expect it to be u64 soon with _realistic_ fee values). You can see this in 
- `apply_to()` returns a `weight`, not a `fee`.
- `make_payment` still operates on this `weight` and internally converts in to `Balance`.

Hence, if you don't have a string objection with the fee not being a generic type which gets glued into `Balances::Currency` somewhere up the line, I'm gonna do the renaming option for this PR.

There is a _propose fees module_ issue open. If things go smooth here I will pursue that in a separate PR.

Reply:
ok - for now let's call it `WeightMultiplier` and extract it to a new module at a later date.